perceval/backends/kitsune.py

```diff
@@ -0,0 +1,326 @@
+
+    @metadata
+    def fetch(self, from_date=DEFAULT_DATETIME):
+        """Fetch questions from the Kitsune url.
+
+        :from_date: obtain questions updated since this date
+        :returns: a generator of questions
+        """
+
+        logger.info("Looking for questions at url '%s' updated after %s",
+                    self.url, str(from_date))
+
+        self._purge_cache_queue()
+
+        nquestions = 0  # number of questions processed
+        tquestions = 0  # number of questions from API data
+        more_updated = True # more questions updated after from_date
+
+        for raw_questions in self.client.get_questions():
+            if not more_updated:
+                break
+            self._push_cache_queue(raw_questions)
+            try:
+                questions_data = json.loads(raw_questions)
+                tquestions = questions_data['count']
+                questions = questions_data['results']
+            except:
+                cause = ("Bad JSON format for mozilla_questions: %s" % (raw_questions))
+                raise ParseError(cause=cause)
+
+            for question in questions:
+                if str_to_datetime(question['updated']) <= from_date:
+                    logging.info("No more questions updated after: %s (last %s)", from_date, question['updated'])
+                    more_updated = False
+                    break
+                question['answers_data'] = []
+                for raw_answers in self.client.get_question_answers(question['id']):
+                    self._push_cache_queue(raw_answers)
+                    answers = json.loads(raw_answers)['results']
+                    question['answers_data'] += answers
+                yield question
+                nquestions += 1
+                self._push_cache_queue('{}') # Mark with empty dict end of question
+
+            logger.debug("Questions: %i/%i", nquestions, tquestions)
+
+            self._flush_cache_queue()
+
+        logger.info("Total number of questions: %i (%i total)", nquestions, tquestions)
+
+    @metadata
+    def fetch_from_cache(self):
+        """Fetch the questions from the cache.
+
+        :returns: a generator of questions
+
+        :raises CacheError: raised when an error occurs accessing the
+            cache
+        """
+
+        logger.info("Retrieving cached questions: '%s'", self.url)
+
+        if not self.cache:
+            raise CacheError(cause="cache instance was not provided")
+
+        cache_items = self.cache.retrieve()
+
+        nquestions = 0
+
+        questions_raw = next(cache_items)
+        while questions_raw:
+            if not json.loads(questions_raw):
+                # Last item is the empty dict
+                questions_raw = next(cache_items)
+            questions = json.loads(questions_raw)['results']
+            for question in questions:
+                question['answers_data'] = []
+                for answers_raw in cache_items:
```

Comment:
The for does automatically the next. Are you thinking in changing it to a while and use next?